[UX] More contrast for texts

andersonjeccel opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 5.x)
* a.b for any bug fixes (e.g. 4.4, 5.1)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/mautic-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

#### Description:

Based on user feedback:

> More contrast: replace the mid gray to a darker gray

The user is suggesting an improvement in the visual contrast of the interface by replacing mid-tone grays with darker shades. This indicates that the current color scheme may not provide sufficient contrast, potentially affecting readability and visual distinction between elements.

The user’s broader need is for a visually accessible and aesthetically pleasing interface that facilitates easy reading and navigation. They are seeking an enhancement in the visual hierarchy that allows users to quickly and effortlessly distinguish between different UI elements.

To address this feedback, this PR adjust the color palette to use darker grays for body text that offer better contrast against other elements. I ensured it meets WCAG accessibility standards for contrast ratios.


![image](https://github.com/mautic/mautic/assets/116097999/d131c6e9-a119-49da-8ea4-93c111ce695d)

The contrast ratio was increased by about ~7 points compared to before.


<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->

#### Steps to test this PR:

<!--
This part is really important. If you want your PR to be merged, take the time to write very clear, annotated and step by step test instructions. Do not assume any previous knowledge - testers may not be developers.
-->
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Look at basically anything, the text will be darker

<!--
If you have any deprecations, list them here along with the new alternative.
If you have any backwards compatibility breaks, list them here.
-->

andersonjeccel committed:
darken

andersonjeccel committed:
grunt compile-less

LordRembo reviewed (approved):
It's an easy fix to do (with no issues as consequence) with an immediate improvement for readability/accessibility, so sounds good to me!

patrykgruszka commented on app/bundles/CoreBundle/Assets/css/app.css:
```diff
@@ -1152,11 +1152,11 @@ h6,
 }
 .text-muted {
-  color: #8393a2;
+  color: #5d6c7c;
```
This change needs to be done in `CoreBundle/Assets/css/app/less/components/type.less:101`.

patrykgruszka commented on app/bundles/CoreBundle/Assets/css/app.css:
```diff
@@ -2912,18 +2912,18 @@ ul.line-legend li {
 }
 .step > li.active .steps {
-  color: #47535f;
+  color: #262d33;
```
LESS file: `app/bundles/CoreBundle/Assets/css/app/less/components/steps.less`

patrykgruszka commented on app/bundles/CoreBundle/Assets/css/app.css:
```diff
@@ -3033,7 +3033,7 @@ ul.line-legend li {
   background-color: #fff !important;
   border-color: #f7f7f7 !important;
-  color: #47535f !important;
+  color: #262d33 !important;
```
LESS: `CoreBundle/Assets/css/app/less/components/colors.less`

patrykgruszka reviewed (changes requested):
The `app.css` file should be generated automatically from the LESS files. You can generate the CSS by executing the `grunt compile-less` command.

andersonjeccel commented:
@patrykgruszka Thanks for pointing!

As I changed 1 variable, every LESS rule or mixin based on the text color also changed accordingly and reflected on CSS when I ran `grunt compile-less`

These different hex codes are a consequence, nothing hardcoded 😉

_(have a look on the commit called “grunt compile-less” for reference)_

patrykgruszka reviewed (approved):
You're right. These variables are dependent on `@text-color`. My earlier comments were not accurate. Looks good 👍

RCheesley committed:
Merge branch '5.x' into ux-More-contrast-replace-the-mid-gray-to-a-darker-gray

RCheesley reviewed (approved):
Thanks for the PR! LGTM! 🚀

andersonjeccel commented:
@RCheesley Based on the approved changes by Patryk and Rembrand and as there are no new commits since their review, can we consider as code-review-passed?

escopecz committed:
Merge branch '5.x' into ux-More-contrast-replace-the-mid-gray-to-a-darker-gray
